tsdb/mapper.go

```diff
@@ -318,13 +391,60 @@ func (lm *SelectMapper) NextChunk() (interface{}, error) {
 			return nil, nil
 		}
 
-		mo := &MapperOutput{}
-		if err := json.Unmarshal(b.([]byte), mo); err != nil {
+		moj := &MapperOutputAsJSON{}
+		if err := json.Unmarshal(b.([]byte), moj); err != nil {
 			return nil, err
-		} else if len(mo.Values) == 0 {
-			// Mapper on other node sent 0 values so it's done.
-			return nil, nil
 		}
+		mvj := []*MapperValueAsJSON{}
+		if err := json.Unmarshal(moj.Values, &mvj); err != nil {
+			return nil, err
+		}
+
+		// Prep the non-JSON version of Mapper output.
+		mo := &MapperOutput{
+			Name:   moj.Name,
+			Tags:   moj.Tags,
+			Fields: moj.Fields,
+		}
+
+		if len(mvj) == 1 && len(mvj[0].AggData) > 0 {
+			// The MapperValue is carrying aggregate data, so run it through the
+			// custom unmarshallers for the map functions through which the data
+			// was mapped.
+			aggValues := []interface{}{}
+			for i, b := range mvj[0].AggData {
+				v, err := lm.mapUnmarshallers[i](b)
```

Comment:
The key fix -- unmarshal the aggregate data coming across the wire into the correct type.